geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateJndiBindingCommand.java

```diff
@@ -165,24 +156,36 @@ else if (alreadyExists)
     if (dsConfigProperties != null && dsConfigProperties.length > 0)
       configuration.setDatasourceConfigurations(Arrays.asList(dsConfigProperties));
 
+    Result result;
+    boolean persisted = false;
     ClusterConfigurationService service = getSharedConfiguration();
 
-    if (service == null) {
-      return ResultBuilder.createUserErrorResult("Cluster Configuration Service is not available. "
-          + "Please connect to a locator with running Cluster Configuration Service.");
+    if (service != null) {
+      boolean alreadyExists = isBindingAlreadyExists(jndiName);
+      if (alreadyExists && ifNotExists)
```

Comment:
if you wait till my PR is done, you can just throw EntityExistException (message, ifNotExist), and the Executor will use that Exception to turn it into appropriate error/info result. Now, you can use EntityNotFoundException, it's the same effect, just the exception name is not very descriptive.